cmd/pctl/main.go

```diff
@@ -155,7 +179,41 @@ func installCmd() *cli.Command {
 				SubName:     subName,
 				Writer:      w,
 			}
-			return catalog.Install(cfg)
+			if err := catalog.Install(cfg); err != nil {
+				return err
+			}
+
+			createPR := c.Bool("create-pr")
+			if createPR {
+				repo := c.String("repo")
+				base := c.String("base")
+				remote := c.String("remote")
+				if repo == "" {
+					return errors.New("repo must be defined if create-pr is true")
+				}
+				fmt.Printf("Creating a PR to repo %s with base %s and branch %s\n", repo, base, branch)
+				r := &git.CLIRunner{}
+				g := git.NewCLIGit(git.CLIGitConfig{
+					Filename: filename,
+					Location: filepath.Dir(filename),
+					Branch:   branch,
+					Remote:   remote,
+					Base:     base,
+				}, r)
+				scmClient, err := git.NewClient(git.SCMConfig{
+					Branch: branch,
+					Base:   base,
+					Repo:   repo,
+				})
+				if err != nil {
+					return fmt.Errorf("failed to create scm client: %w", err)
+				}
+				if err := catalog.CreatePullRequest(scmClient, g); err != nil {
+					return err
+				}
+			}
+
+			return nil
```

Comment:
this `action` func is really long, could some of it be broken out into small funcs :smile: ?

Reply:
Sure.

Reply:
Done